IMAGINGCLOUD-216

s-zubov opened the pull request.

s-zubov committed:
PROBATIONIMAGING-509 - Add Python SDK

s-zubov committed:
Add model template

s-zubov committed:
Add request model template

s-zubov committed:
Add api template

s-zubov committed:
Add python processing to RequestModelExtractor

s-zubov committed:
Add api client template

s-zubov committed:
Add python request models package builder

s-zubov committed:
Update tools binaries

Update RequestModelExtractor to version that processes python. Add
RequestModelPythonPackageBuilder.

s-zubov committed:
Add configuration template

s-zubov committed:
Add rest template

s-zubov committed:
Add .gitignore template

s-zubov committed:
Add __init__package template

s-zubov committed:
Modify request model package builder

s-zubov committed:
Update tools binaries

s-zubov committed:
Update submodule SDKs/Python

s-zubov committed:
Add .swagger-codegen-ignore template

s-zubov committed:
Add python SDK generation script

s-zubov committed:
Remove requirements from ignores

s-zubov committed:
Add requirements template

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Fix request model template

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Fix generatePythonSDK

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Fix generatePythonSDK

s-zubov committed:
Fix generatePythonSDK

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Add gitignore mustache

s-zubov committed:
Fix api template

s-zubov committed:
Add http_request and imaging_request templates

s-zubov committed:
Modify api and api_client templates

s-zubov committed:
Apply codestyle rules

s-zubov committed:
Update license and copyright

s-zubov committed:
Remove noqa comments

s-zubov committed:
Improve code quality for rest template

s-zubov committed:
Improve code quality for api_client template

s-zubov committed:
Improve code quality for configuration template

s-zubov committed:
Improve code quality for api template

s-zubov committed:
Improve code quality for model template

s-zubov committed:
Improve code quality for http_request template

s-zubov committed:
Improve code quality for imaging_request template

s-zubov committed:
Fix api_client template

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Modify SDK generation script

s-zubov committed:
Add SDK spec

s-zubov committed:
Remove SDK spec

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update model.mustache

s-zubov committed:
Update requirements.mustache

s-zubov committed:
Update requirements.mustache

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Fix configuration template

s-zubov committed:
Remove unused imports from model extractor

s-zubov committed:
Add call to Python SDK generation

s-zubov committed:
Add CopyrightFixer

s-zubov committed:
Add description to PythonPackageBuilder

s-zubov committed:
Fix CopyrightFixer binaries

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update generatePythonSDK.bat

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Add headers

s-zubov committed:
Fix template headers

s-zubov committed:
Fix line wrapping for python package

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update copyright

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Fix  RequestModelExtractor

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update requirements

s-zubov committed:
Update python SDK submodule

Commit:
Fix generateSDK.bat

Commit:
Update generatePythonSDK.bat

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Modify api constructor

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Update python SDK submodule

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Add test-requirements to swagger-codegen-ignore

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Fix GET request with empty body

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Update python SDK submodule

s-zubov committed:
Bump version to 19.7

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

MaxCrank commented on codegen/config.python.json:
```diff
@@ -0,0 +1,5 @@
+{
+	"packageName": "asposeimagingcloud",
+	"projectName": "aspose-imaging-cloud",
+	"packageVersion" : "1.0.0"
```
@s-zubov Please, set the appropriate package version.

s-zubov replied:
Thank you, I've updated package version.

MaxCrank commented on codegen/generatePythonSDK.bat:
```diff
@@ -0,0 +1,19 @@
+copy /y Templates\python\.swagger-codegen-ignore %sdkfolder%\.swagger-codegen-ignore
+if exist "%sdkfolder%\asposeimagingcloud\api\" del /S /Q "%sdkfolder%\asposeimagingcloud\api\" || goto :error
+if exist "%sdkfolder%\asposeimagingcloud\models\" del /S /Q "%sdkfolder%\asposeimagingcloud\models\" || goto :error
+java -jar Tools\swagger-codegen-cli-2.3.1.jar generate -i https://api-qa.aspose.cloud/v3.0/imaging/swagger/sdkspec -l python -t Templates\python -o %sdkfolder% -c config.python.json || goto :error
```
@s-zubov Please, make rebase with _development_ branch since it has been modified, and take note on the modifications, especially _swagger-codegen-cli-2.3.1.jar_ replacement with version 2.4.5 which will force modification of Python SDK generation script.

s-zubov replied:
Thank you, I will check what's new in the development branch. Please, notice that modifications related to on-premise deployment are out of scope for this PR.

MaxCrank replied:
@s-zubov Yes, you are absolutely right about the scope.

s-zubov replied:
Thank you, I've fixed swagger-codegen version in SDK generation script and updated copyright message in templates.

MaxCrank commented:
@s-zubov Please, please try to squash the commits for future pull requests as much as possible during your work, e.g. if you have several consequent commits on a single issue, it's better to squash them as soon as the issue is worked out - this way we'll keep the history clean. Thanks.

MaxCrank commented:
@s-zubov Please, modify README.md to add Python SDK to the list of existing SDKs.

s-zubov committed:
Fix version in codegen config

s-zubov committed:
Fix calling wrong version of swagger-codegen

s-zubov committed:
Add reference to Python SDK to README

s-zubov committed:
Update copyright

s-zubov committed:
Modify ApiException

s-zubov committed:
Add deserialization for Error response from the server

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Fix deserialization for Error response from the server

MaxCrank commented:
@s-zubov Please, check if you've rebased or updated the submodules (other SDKs than Python) from _development_ branches for codegen project as well - there should be no code changes in them.

s-zubov committed:
Fix submodules

s-zubov commented:
> @s-zubov Please, check if you've rebased or updated the submodules (other SDKs than Python) from _development_ branches for codegen project as well - there should be no code changes in them.

Thank you, I've moved feature branch in each submodule to the tip of the development branch and updated codegen project.

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Fix deserialization for Error response from the server

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Modify package name to aspose-imaging-cloud
